kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/test/config/ResolvedTestConfig.kt

```diff
@@ -30,13 +32,13 @@ data class ResolvedTestConfig(
     *
     * To set a timeout per invocation see [invocationTimeout].
     */
-   val timeout: Duration?,
+   val timeout: Duration,
 
    /**
     * This timeout applies to individual invocations of a test case. If invocations is 1, then this
     * has the same effect as timeout. To set a timeout across all invocations then see [timeout].
     */
-   val invocationTimeout: Duration?,
+   val invocationTimeout: Duration,
```

Comment:
As I can see it is safe to make these fields non-nullable. The only place where these fields were `null` was `ResolvedTestConfig.default`, used only for Kotest tests and always replaced by a resolved configuration in real test execution.

Reply:
The only problem is the [binary compatibility](https://jakewharton.com/public-api-challenges-in-kotlin/#binary-compatibility) of the data classes. I can provide a secondary constructor, copy method and destructuring operators with old signatures to maintain binary compatibility, but it is a pain.